GVFS/GVFS.Platform.Windows/WindowsPlatform.cs

```diff
@@ -193,27 +193,37 @@ public override string GetNamedPipeName(string enlistmentRoot)
 
         public override void ConfigureVisualStudio(string gitBinPath, ITracer tracer)
         {
-            const string GitBinPathEnd = "\\cmd\\git.exe";
-            string[] gitVSRegistryKeyNames =
+            try
             {
-                "HKEY_CURRENT_USER\\Software\\Microsoft\\VSCommon\\15.0\\TeamFoundation\\GitSourceControl",
-                "HKEY_CURRENT_USER\\Software\\Microsoft\\VSCommon\\16.0\\TeamFoundation\\GitSourceControl"
-            };
-            const string GitVSRegistryValueName = "GitPath";
+                const string GitBinPathEnd = "\\cmd\\git.exe";
+                string[] gitVSRegistryKeyNames =
+                {
+                    "HKEY_CURRENT_USER\\Software\\Microsoft\\VSCommon\\15.0\\TeamFoundation\\GitSourceControl",
+                    "HKEY_CURRENT_USER\\Software\\Microsoft\\VSCommon\\16.0\\TeamFoundation\\GitSourceControl"
+                };
+                const string GitVSRegistryValueName = "GitPath";
 
-            if (!gitBinPath.EndsWith(GitBinPathEnd))
-            {
-                tracer.RelatedWarning(
-                    "Unable to configure Visual Studio’s GitSourceControl regkey because invalid git.exe path found: " + gitBinPath,
-                    Keywords.Telemetry);
+                if (!gitBinPath.EndsWith(GitBinPathEnd))
+                {
+                    tracer.RelatedWarning(
+                        "Unable to configure Visual Studio’s GitSourceControl regkey because invalid git.exe path found: " + gitBinPath,
+                        Keywords.Telemetry);
 
-                return;
-            }
+                    return;
+                }
 
-            string regKeyValue = gitBinPath.Substring(0, gitBinPath.Length - GitBinPathEnd.Length);
-            foreach (string registryKeyName in gitVSRegistryKeyNames)
+                string regKeyValue = gitBinPath.Substring(0, gitBinPath.Length - GitBinPathEnd.Length);
+                foreach (string registryKeyName in gitVSRegistryKeyNames)
+                {
+                    Registry.SetValue(registryKeyName, GitVSRegistryValueName, regKeyValue);
+                }
+            }
+            catch (Exception ex)

```

Comment:
is it worth replacing this catch-all handler with more specific one related to the issue you are fixing (writing to volatile registry keys)

Reply:
Thought about that and I didn't want any exception to stop VFSForGit from mounting since this is purely a helpful thing that we are doing for Visual Studio.